### DISCUSSION TITLE
gpio error msg

### DISCUSSION THREAD

**Discussion by omo22** [tagyoureit/nodejs-poolController#1017] [General]
hi
I am trying to create input gpio and getting error msg, same for output gpio.
any idea why?
![image](https://github.com/user-attachments/assets/91e2e6f9-c6c3-42c0-b4e1-61877bfdc476)
![image](https://github.com/user-attachments/assets/29422380-4ea4-4f16-8ccc-784762590fdf)
Tx

**Comment by rstrouse** [comment]
Did you upgrade your Raspberry Pi to bookworm?  If you did you need to set the board type to bookworm os.

![image](https://github.com/user-attachments/assets/251102d7-ee54-48ed-8af5-c85a53b6893e)

**Reply by omo22** [reply]
yes i did  now working but except  pin 7 still not working
Tx alot 

![image](https://github.com/user-attachments/assets/97349612-1872-4cfa-9cdb-cd7e8a1ed13e)

e.g pin 8 is ok 
![image](https://github.com/user-attachments/assets/6107ff8b-c9e3-4372-bab4-10d070d55e70)

**Reply by rstrouse** [reply]
Pin #7 on the RPi is one of those pins that have a pullup enabled by default making them an output pin.  This can be changed using dtoverlay settings in the config.txt file.  I don't know if this became goofier with the advent of Bookworm.

**Comment by omo22** [comment]
thanks
I want to have GPIO 10 trigger a relay on off
but there are no options to do so
any idea?
![image](https://github.com/user-attachments/assets/e98e4584-26b0-4d7b-bcc8-62c3140deb03)
